### DISCUSSION TITLE
grad / jacobian over a subset of a pytree

### DISCUSSION THREAD

**Discussion by gerdm** [jax-ml/jax#12376] [Q&A]
Is there a way to compute the gradient / jacobian of a function that takes a pytree w.r.t a subset of the pytree keys? For instance, suppose I have 

```python
X0 = {
    "w0": 3.0,
    "w1": 1.0,
    "w2": 10.0
}

def g(...):
    # Some computationally-expensive function that I do not want to differentiate over
    ...

def f(X):
    return 3 * X["w0"] + X["w1"] ** 2 * g(X["w2"] ** 2)
```

I would like to compute `jax.grad(f)(X0)`, but only over `"w0"` and `"w1"`. I understand that a way to do this would be to declare a helper function `f_help(X0_subset, w2)` that takes a subset of the pytree for and the value `"w2"` outside of the pytree and then re-create the input pytree, but I'm wondering if there's a _cleaner_ way to do this.

Thanks!
